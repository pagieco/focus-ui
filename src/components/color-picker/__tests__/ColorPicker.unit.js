import { shallowMount } from '@vue/test-utils';
import { ColorPicker } from '../index';

const shallowWrap = (propsData = {}) => shallowMount(ColorPicker, { propsData });

describe('ColorPicker', () => {
  it('is a valid vue component', () => {
    expect(ColorPicker).toBeAComponent();
  });

  it('matches the snapshot', () => {
    expect(shallowMount(ColorPicker).element).toMatchSnapshot();
  });

  it('uses the allow-alpha prop', () => {
    const wrapper = shallowWrap({ allowAlpha: false });

    expect(wrapper.find('.color-picker-alpha').exists()).toBeFalsy();

    wrapper.setProps({ allowAlpha: true });

    expect(wrapper.find('.color-picker-alpha').exists()).toBeTruthy();
  });

  it('returns a tinycolor object when retrieving the current-color', () => {
    const wrapper = shallowWrap({ value: 'red' });

    const color = wrapper.vm.currentColor;

    expect(color).toBeInstanceOf(Object);
    expect(color).toMatchObject({
      _originalInput: 'red',
      _r: 255,
      _g: 0,
      _b: 0,
    });
  });

  it('correctly sets the current-color to a new color through tinycolor and triggers the input event', () => {
    const wrapper = shallowWrap();
    wrapper.vm.currentColor = 'blue';

    const color = wrapper.vm.currentColor;

    expect(color.toRgb().b).toBe(255);
    expect(wrapper.emitted('input')).not.toBeUndefined();
  });

  it('gets the saturation', () => {
    const wrapper = shallowWrap({ value: 'blue' });

    expect(wrapper.vm.saturation).toBe('hsl(240, 100%, 50%)');
  });

  it('can set the saturation', () => {
    // TODO
  });

  it('can set the hue', () => {
    // TODO
  });

  it('can set the alpha', () => {
    // TODO
  });

  it('can update the saturation dragger position', () => {
    // TODO
  });

  it('can update the hue ruler position', () => {
    // TODO
  });

  it('can update the alpha ruler position', () => {
    // TODO
  });
});
