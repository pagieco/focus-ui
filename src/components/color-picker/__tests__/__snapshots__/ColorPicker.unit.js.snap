// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker matches the snapshot 1`] = `
<div
  class="color-picker"
>
  <div
    class="color-picker-saturation"
  >
    <div
      class="color-picker-saturation__color"
    />
     
    <div
      class="color-picker-saturation__black"
    />
     
    <div
      class="color-picker-saturation__dragger"
    />
  </div>
   
  <div
    class="color-picker-hue"
  >
    <div
      class="color-picker-hue__ruler"
      style="top: 100%;"
    />
  </div>
   
  <div
    class="color-picker-alpha"
  >
    <div
      class="color-picker-alpha__black"
    />
     
    <div
      class="color-picker-alpha__ruler"
      style="top: 100%;"
    />
  </div>
</div>
`;
